File: src/db/models/letter.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Letter = sequelize.define('Letter', {
    senderId: DataTypes.INTEGER,
    countryId: DataTypes.INTEGER,
    userOwlId: DataTypes.INTEGER,
    receiverId: DataTypes.INTEGER,
    content: DataTypes.TEXT,
    sentDate: DataTypes.DATE,
    deliveryDate: DataTypes.DATE,
    pickUpDate: DataTypes.DATE
  }, {});
  Letter.associate = function(models) {
    // associations can be defined here
    Letter.belongsTo(models.User, {as: 'sender'})
    Letter.belongsTo(models.Country)
    Letter.belongsTo(models.UserOwl)
    Letter.belongsTo(models.User, {as: 'receiver'})
  };
  return Letter;
};

File: app/models/letter.rb
class Letter < ApplicationRecord
  belongs_to :user_owl
  belongs_to :user, :foreign_key => 'sender_id'
  belongs_to :user, :optional => true, :foreign_key => 'receiver_id'
  belongs_to :country, :foreign_key => 'from_country_id'
  belongs_to :country, :foreign_key => 'to_country_id'

  after_initialize :init

  def init
    self.sent_date = Time.now if self.sent_date.nil?
  end 
end
